Generate .iex.exs files with phx.new. alias MyApp.Repo for Ecto apps.

TheFirstAvenger opened the pull request:
`.iex.exs` files are a very useful feature of Elixir, as all commands in them are run when iex session is started from the folder containing the file. This is a great place for aliases. I often find Elixir Engineers that don't know about this feature, so adding them to the generators not only eliminates a step for those of us who already use them them, but it also helps others learn about the feature.

This PR modifies `mix phx.new` to generate an `.iex.exs` file at the root of the project being generated, including a short commented explanation of the file and what it does. Additionally, when you are creating a project with Ecto it injects `alias MyApp.Repo` in the `.iex.exs` files. In an umbrella app, it will inject the repo alias both at the app root and the umbrella root.

aaronrenner reviewed (commented):
Good idea, @TheFirstAvenger! Would it also make sense to add a `.iex.exs` to the web project in the umbrella?

Also, it may be worth adding a note about this file to the guides in this or a follow up PR.

TheFirstAvenger commented on installer/lib/phx_new/umbrella.ex:
```diff
@@ -5,6 +5,7 @@ defmodule Phx.New.Umbrella do
 
   template :new, [
     {:eex,    "phx_umbrella/gitignore",               :project, ".gitignore"},
+    {:eex,    "phx_umbrella/.iex.exs",                :project, ".iex.exs"},
```
Good catch, fixed!

TheFirstAvenger commented:
@aaronrenner

>Would it also make sense to add a .iex.exs to the web project in the umbrella?

I was actually leaning the other way, thinking about removing the one in the ecto app directory to reduce the amount of files generated (I know there is a larger concern about the amount of files generated by the generator). In my workflow with umbrella projects, I never run iex from anything but the root of the umbrella. I wonder if that is normal or if there are people who regularly run IEx from the apps directories. I had only originally left the Ecto one in because I ran into an issue determining if the command was generating an app by being run inside the apps directory vs if this was a --umbrella app command that was creating the sub-app. I have that figured out now, so I can go either way - add it to the web directory or remove it from the ecto one. Thoughts?

>Also, it may be worth adding a note about this file to the guides in this or a follow up PR.

Good idea. I have added a section to the ecto.md guide where we tell the user to `alias Hello.User` in the IEx session, noting how the alias can alternatively be added to the `.iex.exs` file. I didn't see another great place to add it, but let me know if you think there is an alternative or additional place it could go.

aaronrenner commented:
> In my workflow with umbrella projects, I never run iex from anything but the root of the umbrella.

Now that I think about it, I usually only run iex from the root of the umbrella too. 😁 I agree with the argument to only have one in an umbrella so the user doesn't get confused which one is running.

> I have added a section to the ecto.md guide where we tell the user to alias Hello.User in the IEx session, noting how the alias can alternatively be added to the .iex.exs file. I didn't see another great place to add it, but let me know if you think there is an alternative or additional place it could go.

I agree it doesn't seem like there's a great place to add it. The first place I thought was in the Directory Structure guide, but it doesn't cover any individual files. We could also take the approach of just letting the user discover it, but it's a hidden file so that might be tough. 😛 

@josevalim Any thoughts on either of these?

TheFirstAvenger commented:
@aaronrenner Another thought I just had was if it might make sense to add `IO.puts "alias MyApp.Repo"` to the `.iex.exs` file 
 in addition to actually running the alias command, as currently these commands all happen silently to the end user when IEx is run. This would bring more explicitness to the actions we are automatically adding.

josevalim commented:
I delegate the decision to @chrismccord  :)

TheFirstAvenger commented:
@aaronrenner I have removed the entries inside the umbrella sub-apps, so now on both umbrellas and single apps only one file is created at the root of the project. I also added IO.puts to the file [per this comment from José](https://github.com/elixir-lang/elixir/pull/10634#issuecomment-756944279).

chrismccord commented:
Thanks for hanging tight on this mike! I agree .iex.exs is super nice to have in projects, but we currently balance the number of files we generate vs how many are used. I use .iex.exs in all of my phoenix projects, but one of the concerns is folks not knowing it is there in the first place, so I am unsure about having it as a default, even tho it's a great feature :) Thanks!
